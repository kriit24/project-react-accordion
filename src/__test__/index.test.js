// test/script.js
import React from 'react';
import {View, Text} from 'react-native';
import renderer from 'react-test-renderer';
import expect from "expect";
import Accordion from "../index";

let style = {
    accordion: {
        borderRadius: 5,
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: '#ced4da',
        color: 'grey',
        direction: 'ltr',
        padding: 0,
        marginBottom: 10
    },
    heading: {},
    list: {},
};

export default class MainApp extends React.Component {

    render() {

        let title = <Text>TITLE</Text>;

        return (
            <Accordion title={title} key={Math.random()} style={style} open="one" duration="150" visible={false}>
                <View style={{padding: 5}}>
                    <Text>CONTENT</Text>
                </View>
            </Accordion>
        );
    }
}

test('renders correctly', () => {
    //let tree = Test('see');
    //console.log(tree);
    const tree = renderer.create(<MainApp />).toJSON();
    expect(tree).toMatchSnapshot();
});
