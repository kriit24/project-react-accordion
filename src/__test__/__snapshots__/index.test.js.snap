// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "backgroundColor": "#FFFFFF",
        "padding": 5,
        "width": "100%",
      },
      {
        "borderColor": "#ced4da",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "grey",
        "direction": "ltr",
        "marginBottom": 10,
        "padding": 0,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "overflow": "hidden",
      }
    }
  >
    <Text>
      TITLE
    </Text>
  </View>
  <View
    style={
      [
        {},
        {
          "height": 0,
          "padding": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        {
          "overflow": "hidden",
        },
      ]
    }
  />
</View>
`;
